src/OSPSuite.UI/Services/DialogCreator.cs

```diff
@@ -79,24 +83,46 @@ public ViewResult MessageBoxYesNo(string message, ViewResult defaultButton = Vie
 
       public ViewResult MessageBoxYesNo(string message, string yes, string no, ViewResult defaultButton = ViewResult.Yes)
       {
-         return showQuestionMessageBox(message, new [] {
-            DialogResult.Yes, DialogResult.No}, yes, no, string.Empty, defaultButton);
+         return showQuestionMessageBox(message, new[]
+         {
+            DialogResult.Yes, DialogResult.No
+         }, yes, no, string.Empty, defaultButton);
+      }
+
+      public ViewResult MessageBoxConfirm(string message, Action doNotShowAgain, ViewResult defaultButton = ViewResult.Yes)
+      {
+         return showQuestionMessageBox(message, new[]
+         {
+            DialogResult.Yes, DialogResult.Cancel
+         }, string.Empty, string.Empty, string.Empty, defaultButton, doNotShowAgain);
       }
 
-      private ViewResult showQuestionMessageBox(string message, IReadOnlyList<DialogResult> buttons, string yes, string no, string cancel, ViewResult defaultButton)
+      private ViewResult showQuestionMessageBox(string message, IReadOnlyList<DialogResult> buttons, string yes, string no, string cancel, ViewResult defaultButton, Action doNotShowAgain = null)
       {
          var currentLocalizer = Localizer.Active;
          try
          {
             Localizer.Active = new XtraMessageBoxLocalizer(yes, no, cancel);
-            return _mapper.MapFrom(XtraMessageBox.Show(createMessageBoxArgs(message, buttons.ToArray(), getIcon(SystemIcons.Question), defaultButtonFrom(defaultButton))));
+            var args = createMessageBoxArgs(message, buttons.ToArray(), getIcon(SystemIcons.Question), defaultButtonFrom(defaultButton), doNotShowAgain != null);
+            args.Closed += (sender, e) => onMessageBoxClosed(e, doNotShowAgain);
+            DialogResult input = XtraMessageBox.Show(args);
+            return _mapper.MapFrom(input);
          }
          finally
          {
             Localizer.Active = currentLocalizer;
          }
       }
 
+      private void onMessageBoxClosed(XtraMessageBoxClosedArgs e, Action doNotShowAgain)
+      {
+         // e.Visible gets whether the user checked the 'Do not show this message again' checkbox. Checked makes e.Visible = false 
```

Comment:
I don't get it. 
Why does `e.Visible` correspond to the check status of the 'Do not show this message again' checkbox?

Reply:
I think you'd have to ask the DevExpress guys why it's named that

https://docs.devexpress.com/WindowsForms/DevExpress.XtraEditors.XtraMessageBoxEventArgs.Visible

Reply:
Strange.. but OK